docs/reference/glossary/network-latency-management.md

```diff
@@ -0,0 +1,112 @@
+ 
+
+## Interpolation
+
+A ------> B
+
+Interpolation works by essentially buffering a game state and rendering the game state to the player with a small, constant delay. When a packet from the server arrives, before immediately updating the position of an object , the client will interpolate the position, starting from the last known position. 
+
+Over an interpolation interval, the object will be rendered  smoothly  moving between the two positions. 
+
+## Extrapolation
+
+
+A ---- B -----> 
+
+Extrapolation is an attempt to estimate a future game state. On receipt of a packet from the server, the position of an object is updated to the new position. Awaiting the next update, the next position is extrapolated based on the current position and the movement at the time of the update. 
+
+The client will normally assume that a moving object will continue in the same direction. When a new packet is received, the position may be updated.
+
+## Action casting
+
+Server driven action, but with client-side “casting” animations and sounds that respond to client inputs. This is to get responsiveness while still keeping the action server driven.
+
+## Client prediction:
+
+- **Deterministic lockstep:**
+  A method of networking a system from one computer to another by sending only the inputs that control that system, rather than the state of that system
+
+- **Deterministic rollback:**
+  An enhancement of deterministic lockstep where clients forward-predict inputs while waiting for updates. This setup is more complex but enables a more responsive game than lockstep. It’s relatively inexpensive and secure, but with very complex determinism and simulation.
+
+- **Input prediction:**
+  (and reconciliation to correct differences between client and server) TODO
+
+- **Ghost prediction**
+  Client side prediction where input isn’t involved. For example, knowing an AI’s state at frame i, we can predict its state at time i+1 assuming it’s deterministic enough to run the same both client side and server side.
+     
+
+## Latency
+
+Is the time delay between the cause and the effect of some physical change in the system being observed. 
+
+Network time between sending a packet and receiving that packet.
+
+## Lag
+
+Perception felt by an end user that there are delays to their inputs, often caused by network latency. 
+
+## Jitter
+
+The rate at which ping changes over a period of time (otherwise known as 'ping spikes' or 'stuttering'). 'Jittering' is essentially the fluctuation or variation of latency over time
+
+## RTT
+
+Latency A → B + frame calculation time + B → A
+
+Is the amount of time it takes for a signal to be sent plus the amount of time it takes for an acknowledgement of that signal to be received. 
+
+## Ping
+
+A → B + B → A
+
+Round Trip Time without frame calculation. 
```

Comment:
These are not clearly defined. Ping and RTT are interchangeable we shouldn't put hard definitions on them

Reply:
Note added to clarify sections are a general guide to meanings and  not hard definitions